Source/Adobe.Target.Client/Util/CookieUtils.cs

```diff
@@ -0,0 +1,140 @@
+namespace Adobe.Target.Client.Util
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using Adobe.Target.Client.Model;
+
+    /// <summary>
+    /// Cookie Utils
+    /// </summary>
+    internal static class CookieUtils
+    {
+        private const string CookieValueSeparator = "|";
+        private const string InternalCookieSerializationSeparator = "#";
+        private const int SessionIdCookieMaxAge = 1860;
+        private const int DeviceIdCookieMaxAge = 63244800;
+        private const int ClusterLocationHintMaxAge = 1860;
+
+        /// <summary>
+        /// Parse Target cookie
+        /// </summary>
+        /// <param name="targetCookie">Target cookie</param>
+        /// <returns>Parsed Target cookies</returns>
+        internal static Dictionary<string, string> ParseTargetCookie(string targetCookie)
+        {
+            if (string.IsNullOrEmpty(targetCookie))
+            {
+                return new Dictionary<string, string>();
+            }
+
+            var nowInSeconds = (int)(DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() / 1000);
+
+            return targetCookie.Split(CookieValueSeparator)
+                .TakeWhile(cookie => !string.IsNullOrEmpty(cookie))
+                .Select(DeserializeInternalCookie)
+                .Where(internalCookie => internalCookie != null && internalCookie.MaxAge > nowInSeconds)
+                .ToDictionary(internalCookie => internalCookie.Name, internalCookie => internalCookie.Value);
+        }
+
+        /// <summary>
+        /// Gets Location Hint from Tnt Id
+        /// </summary>
+        /// <param name="tntId">Tnt Id</param>
+        /// <returns>Location hint</returns>
```

Comment:
Could we use a nullable type so it is explicit in the method signature.

Reply:
Fixed